src/main/flight/position.c

```diff
@@ -129,23 +144,39 @@ if (sensors(SENSOR_GPS) && STATE(GPS_FIX)) {
     baroAlt -= baroAltOffset;
     gpsAlt -= gpsAltOffset;
     
-    if (haveGpsAlt && haveBaroAlt) {
-        estimatedAltitudeCm = gpsAlt * gpsTrust + baroAlt * (1 - gpsTrust);
+    if(positionConfig()->altSource == BARO_ONLY){
```

Comment:
The build fails because you are not checking `haveBaroAlt` here - this evaluates as constant false if `USE_BARO` isn't defined, causing the compiler to drop the entire block, and not tripping over missing `barometerConfig` on targets with no BARO support.

This should also be done for `haveGpsAlt` below, and the entire condition should probably be rewritten to check only once, and not once in the outer `if`, and then again in the inner `if`.

Also, the `#undef USE_VARIO` in `src/main/target/common_post.h` should only be conditional to `USE_BARO` to correct the dependency on `barometerConfig` that is tripping up above. But this can be fixed in another pull request.

Reply:
got it. i'll fix it. thanks